pkg/imagestore/imagestore.go

```diff
@@ -114,15 +134,19 @@ func downloadURLToFile(url string, path string) error {
 }
 
 func (s *rhcosStore) Populate(ctx context.Context) error {
+	if err := s.cleanDataDir(); err != nil {
+		return err
```

Comment:
Will this make the service fail? Or is it just a runtime error? (not super familiar with the code base, trying to learn)

A more philosophical question:

Should the service be considered healthy if the `Populate` function fails (whether it's removing or writing files)? (maybe this is already handled)

Reply:
> Will this make the service fail?

Right now, [yes](https://github.com/openshift/assisted-image-service/blob/24cda2dcc77cbf2dc0bcb601b41b8391d5cceccc/main.go#L56-L59). I struggled with this one for a while.

> Should the service be considered healthy if the Populate function fails

If the deletes fail ... maybe, but we cannot consider it healthy if the writes don't succeed.

Reply:
I think that it should fail if we have issues with the storage we should not hide this issue

Reply:
I think we'll keep this as is then. I don't expect to have too many problems with this.